modules/peers.js

```diff
@@ -286,6 +287,31 @@ Peers.prototype.sandboxApi = function (call, args, cb) {
 	sandboxHelper.callMethod(Peers.prototype.shared, call, args, cb);
 };
 
+/**
+ * Calculates consensus for as a ratio active to matched peers.
+ * @param {Array<Peer>} matched - peers with same as system broadhash
+ * @param {Array<Peer>} active - active peers (with connected state)
+ * @returns {number|undefined} - return consensus or undefined if config.forging.force = true
+ */
+Peers.prototype.getConsensus = function (matched, active) {
+
+	if (library.config.forging.force) {
+		return undefined;
+	}
+
+	active = active || __private.getByFilter({state: Peer.STATE.CONNECTED, normalized: false});
```

Comment:
`__private.getByFilter` is heavy (we don't need things like random sorting here and checking allowed fields), please use `library.logic.peers.list(true)` to get peers.